packages/loot-core/src/server/importers/ynab5.ts

```diff
@@ -152,16 +152,134 @@ async function importTransactions(
   const payeeTransferAcctHashMap = new Map<string, YNAB5.Payee>(
     payeesByTransferAcct,
   );
+  const orphanTransferMap = new Map<string, YNAB5.Transaction[]>();
+  const orphanSubtransfer = [] as YNAB5.Subtransaction[];
+  const orphanSubtransferTrxId = [] as string[];
+  const orphanSubtransferAcctIdByTrxIdMap = new Map<string, string>();
+  const orphanSubtransferDateByTrxIdMap = new Map<string, string>();
 
   // Go ahead and generate ids for all of the transactions so we can
   // reliably resolve transfers
-  for (const transaction of data.transactions) {
+  // Also identify orphan transfer transactions and subtransactions.
+  for (const transaction of data.subtransactions) {
     entityIdMap.set(transaction.id, uuidv4());
+
+    if (transaction.transfer_account_id) {
+      orphanSubtransfer.push(transaction);
+      orphanSubtransferTrxId.push(transaction.transaction_id);
+    }
   }
-  for (const transaction of data.subtransactions) {
+
+  for (const transaction of data.transactions) {
     entityIdMap.set(transaction.id, uuidv4());
+
+    if (
+      transaction.transfer_account_id &&
+      !transaction.transfer_transaction_id
+    ) {
+      const key =
+        transaction.account_id + '#' + transaction.transfer_account_id;
+      if (!orphanTransferMap.has(key)) {
+        orphanTransferMap.set(key, [transaction]);
+      } else {
+        orphanTransferMap.get(key).push(transaction);
+      }
+    }
+
+    if (orphanSubtransferTrxId.includes(transaction.id)) {
+      orphanSubtransferAcctIdByTrxIdMap.set(
+        transaction.id,
+        transaction.account_id,
+      );
+      orphanSubtransferDateByTrxIdMap.set(transaction.id, transaction.date);
+    }
   }
 
+  // Compute link between subtransaction transfers and orphaned transaction
+  // transfers. The goal is to match each transfer subtransaction to the related
+  // transfer transaction according to the accounts, date, amount and memo.
+  const orphanSubtransferMap = orphanSubtransfer.reduce(
+    (map, subtransaction) => {
+      const key =
+        subtransaction.transfer_account_id +
+        '#' +
+        orphanSubtransferAcctIdByTrxIdMap.get(subtransaction.transaction_id);
+      if (!map.has(key)) {
+        map.set(key, [subtransaction]);
+      } else {
+        map.get(key).push(subtransaction);
+      }
+      return map;
+    },
+    new Map<string, YNAB5.Subtransaction[]>(),
+  );
+
+  const orphanTransferComparator = (
+    a: YNAB5.Transaction | YNAB5.Subtransaction,
+    b: YNAB5.Transaction | YNAB5.Subtransaction,
+  ) => {
+    const date_a =
+      'date' in a
+        ? a.date
+        : orphanSubtransferDateByTrxIdMap.get(a.transaction_id);
+    const date_b =
+      'date' in b
+        ? b.date
+        : orphanSubtransferDateByTrxIdMap.get(b.transaction_id);
+    const amount_a = 'date' in a ? a.amount : -a.amount;
+    const amount_b = 'date' in b ? b.amount : -b.amount;
+
+    if (date_a > date_b) return 1;
```

Comment:
worth a quick inline comment for future readers to outline the priority of comparison, so they don't have to read the returns

Reply:
Of course! I added comments.